### PR TITLE
Remove extraneous dispose

### PR DESCRIPTION
**Pull request by yonihemi**
`LLVMGetTargetName` doesn't return a newly allocated string. This crashes under certain circumstances.

### CODE REVIEW

**Review by dabrahams — CHANGES_REQUESTED** [review]
Nice, and thanks!  Could you please add a test that would fail or crash without your change?

File: Tests/LLVMTests/TargetTests.swift [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import SwiftyLLVM
+import XCTest
+
+final class TargetTests: XCTestCase {
+
+  func testHostTargetName() throws {
+    let h = try Target.host()
+    let n = h.name
```
Comment (dabrahams) [inline Tests/LLVMTests/TargetTests.swift:8]: A comment here explaining what potential bug this tests for would be nice, because on the face of it, it just looks like it's testing the target name function, which it isn't really doing—it doesn't check that the name is correct in any way.  In fact, the name of the test (and the `XCTestCase`, and the file) should be adjusted to reflect what we're really testing for.

**Review by dabrahams — CHANGES_REQUESTED** [review]
We're very close, thanks for your work so far!

Did you verify that the test fails for you locally without the rest of the change?

**Comment by yonihemi** [comment]
> Did you verify that the test fails for you locally without the rest of the change?

It actually crashes the test with a SIGABRT.
[asciicast](https://asciinema.org/a/OXpynmWNEmrp3AnythKDCE6qb)
I've tried catching the signal and turn it into a XCTFail but that deosn't seem to work in a test runner.
Originally I've hit this crash with some more meaningful changes (I have some basic Hylo cross-compiling working) but even the simple test case reproduces the crash.

File: Tests/LLVMTests/TargetTests.swift (hunk 1 above)
Comment (yonihemi) [reply Tests/LLVMTests/TargetTests.swift:8]: Agreed about this test, but I think there is merit in having a seperate test case for the `Target` struct.
For host name we could e.g. shell to `uname` or use compiler conditionals, but it will be a big job verifying all supported platforms. For now a test of a non-empty value could be useful, perhaps with a better name.
How about a test case test like this? (I hate the actual function names though)
```swift
final class TargetTests: XCTestCase {

  /// Accessing the `name` property previously crashed the process.
  func testAccessingTargetName() throws {
    let h = try Target.host()
    _ = h.name
  }
  
  func testIsHostTargetNameValid() throws {
    let h = try Target.host()
    let n = h.name
    XCTAssertFalse(n.isEmpty)
  }

}
```

Comment (kyouko-taiga) [reply Tests/LLVMTests/TargetTests.swift:8]: I don't think the docstring of the first method should say that the method exists because of a past bug. That's an inline comment I would put in the function itself or I would just add a note in the commit that fixed the issue.

I don't mind if we don't have one test per solved issue as long as fixes _are_ testes. So I would be satisfied with only the second test (i.e., what your PR proposes). As for the name/docs I would simply write "testNonEmptyHostTarget".

In summary, I would offer:
```swift
func testNonEmptyHostTarget() throws {
  let h = try Target.host()
  let n = h.name
  XCTAssertFalse(n.isEmpty)
}

**Review by kyouko-taiga — APPROVED** [review]
Everything looks good to me except for the minor comment I added to the discussion about the name of the test.

File: Tests/LLVMTests/TargetTests.swift (hunk 1 above)
Comment (dabrahams) [reply Tests/LLVMTests/TargetTests.swift:8]: > I don't think the docstring of the first method should say that the method exists because of a past bug.
Yes, tests normally don't get docstrings, but if they did, the docstring should describe what the test does, just  like for any other function.

> As for the name/docs I would simply write "testNonEmptyHostTarget".

I disagree with that. I specifically asked for a name reflecting the regression being tested for; i.e. that the lifetime is properly handled—which in this case, based on the implementation, would be amount to “can the name be accessed at all?”

A test for the target struct is a useful thing on its own, sure, but as written this looks like a test of the underlying LLVM functionality, which we have no reason to do.

Comment (yonihemi) [reply Tests/LLVMTests/TargetTests.swift:8]: I appreciate your patience and all the hints, but am unable to find a satisfactory solution. 
Hopefully you can fix the issue and I can learn from that. 
Thanks both for your time.

File: Tests/LLVMTests/TargetTests.swift [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import SwiftyLLVM
+import XCTest
+
+final class TargetTests: XCTestCase {
+
+  func testHostTargetName() throws {
```
Comment (dabrahams) [inline Tests/LLVMTests/TargetTests.swift:6]:
```suggestion
  func testHostTargetNameCopyValidity() throws {
    // Check that a copy of the target name can still be used after the lifetime
    // of the Target ends.
```

**Review by dabrahams — APPROVED** [review]

**Comment by dabrahams** [comment]
Thanks for your worn on this @yonihemi; I'm very sorry if we discouraged you.
